ui/src/app/model/portCall/portcall-timestamp-type.enum.ts

```diff
@@ -18,3 +20,43 @@ export enum PortcallTimestampType {
   PTD_Berth = 'PTD-Berth',
   ATD_Berth = 'ATD-Berth'
 }
+
+export class TimestampTypeUtil {
+
+  static getPortcallTimestampTypes(publisherRole: PublisherRole): string[] {
+    switch (publisherRole) {
+      case PublisherRole.CA:
+        return [
+          PortcallTimestampType.ETA_Berth,
+          PortcallTimestampType.PTA_Berth,
+          PortcallTimestampType.ETA_PBP,
+          PortcallTimestampType.PTA_PBP,
+          PortcallTimestampType.ATA_PBP,
+          PortcallTimestampType.ATA_Berth,
+          PortcallTimestampType.RTC_Cargo_Ops,
+          PortcallTimestampType.ETD_Berth,
+          PortcallTimestampType.PTD_Berth,
+          PortcallTimestampType.ATD_Berth,
+        ]
+      case PublisherRole.TR:
+        return [
+          PortcallTimestampType.RTA_Berth,
+          PortcallTimestampType.ATS_Cargo_Ops,
+          PortcallTimestampType.ETC_Cargo_Ops,
+          PortcallTimestampType.PTC_Cargo_Ops,
+          PortcallTimestampType.ATC_Cargo_Ops,
+        ]
+      case PublisherRole.ATH:
+        return [
+          PortcallTimestampType.RTA_PBP,
+          PortcallTimestampType.RTD_Berth,
+        ]
+      case PublisherRole.PLT:
+        return [
+          PortcallTimestampType.ATS_Pilot,
+        ]
+      default:
+        return [];
+    }
+  }
```

Comment:
@AndreasHoff:  Could you please go through this and match it with the most recent table we got from Guido? (To confirm that those rules are the `Sender` of those timestamps).

(It looks correct at first glance)

Reply:
Everything seems to be in order @nt-as ! :)